pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java

```diff
@@ -283,4 +283,52 @@ private static String getGroupIdFromRealtimeDataTable(String realtimeTableName,
     }
     return groupId;
   }
+
+  private static class KafkaPartitionsCountFetcher implements Callable<Boolean> {
+    private int _partitionCount = -1;
+    private final KafkaStreamMetadata _kafkaStreamMetadata;
+    private Exception _exception;
+
+    private KafkaPartitionsCountFetcher(KafkaStreamMetadata kafkaStreamMetadata) {
+      _kafkaStreamMetadata = kafkaStreamMetadata;
+    }
+
+    private int getPartitionCount() {
+      return _partitionCount;
+    }
+
+    private Exception getException() {
+      return _exception;
+    }
+
+    @Override
+    public Boolean call() throws Exception {
```

Comment:
No, we actually want to create a new SimpleConsumerWrapper if it throws transient exception. (Currently) it does not reset its state correctly.